frontend/packages/kubevirt-plugin/src/components/modals/disk-modal/disk-modal.tsx

```diff
@@ -172,28 +171,26 @@ export const DiskModal = withHandlePromise((props: DiskModalProps) => {
     type,
   });
 
-  const resultDataVolumeName = prefixedID(vmName, name);
-
   const resultVolume = VolumeWrapper.initializeFromSimpleData({
     name,
     type: source.getVolumeType(),
     typeData: {
-      name: resultDataVolumeName,
+      name: name,
       claimName: pvcName,
       image: containerImage,
     },
   });
 
   let resultDataVolume;
   if (source.requiresDatavolume()) {
-    resultDataVolume = DataVolumeWrapper.initializeFromSimpleData({
-      name: resultDataVolumeName,
-      storageClassName: storageClassName || null, // || null is to enable merging
-      type: source.getDataVolumeSourceType(),
-      size,
-      unit,
-      typeData: { name: pvcName, namespace, url },
-    })
+    resultDataVolume = new DataVolumeWrapper()
+      .init({
+        name: name,
```

Comment:
shouldn't this be reusing the old DataVolume name if such exists?

Reply:
and generating the rand suffix when you are not using the wizard?

Reply:
Done!